layer/nrlf/core/validators.py

```diff
@@ -318,6 +291,47 @@ def _validate_ssp_asid(self, model: DocumentReference):
             )
             return
 
+    def _validate_ssp_asid(self, model: DocumentReference):
+        """
+        Validate that the document contains a valid ASID in the context.related field when the content contains an SSP URL
+        """
+
+        ssp_content = any(
+            [
+                content
+                for content in model.content
+                if content.attachment.url.startswith("ssp://")
+            ]
+        )
+
+        logger.log(LogReference.VALIDATOR001, step="ssp_content_and_asid_exists")
+
+        does_related_exist = getattr(model.context, "related", None)
+        does_asid_exist = False
+        if does_related_exist:
+            asid_references = [
+                (idx, related)
+                for idx, related in enumerate(getattr(model.context, "related", []))
+                if related.identifier.system == "https://fhir.nhs.uk/Id/nhsSpineASID"
+            ]
+            does_asid_exist = True
+            self._validate_asid(asid_references)
```

Comment:
I think the new logic has introduced an unwanted validation error if you provide `context.related` but no ASID, you can verify it with this unit test:

```
def test_validate_with_content_related_but_no_asid():
    validator = DocumentReferenceValidator()
    document_ref_data = load_document_reference_json(
        "Y05868-736253002-Valid"
    )

    document_ref_data["context"]["related"] = [
        {
            "identifier": {
                "system": "https://fhir.nhs.uk/Id/ods-organization-code",
                "value": "Y05868",
            }
        }
    ]

    result = validator.validate(document_ref_data)

    assert result.is_valid is True
```